Add Affiliate-B support for amp-ad

Affiliate-B opened the pull request:
We want to add Affiliate-B Ad Network into the supported ad network for
amp-ad.

We have signed Google Corporate CLA.

Corporation name: For it inc.

Thank you.

jridgewell commented:
/to @lannka

lannka commented:
I saw you are on a quite old branch. Could you please sync to HEAD and resolve the conflicts?
And please follow the [Developer guidelines for a pull request](https://github.com/ampproject/amphtml/blob/master/ads/README.md#developer-guidelines-for-a-pull-request), and implement `render-start` API.

Affiliate-B commented:
Thank you for your reply.

We added render-start API, and fixed the merge conflicts.

Please let me know if there is anything else you'd like me to look at.

Affiliate-B commented:
@lannka 
Could you please review and advise?

Affiliate-B commented on ads/_config.js:
```diff
@@ -20,8 +20,88 @@
  *
  * yourNetworkName: {  // This is the "type" attribute of <amp-ad>
  *
+<<<<<<< HEAD
```
Thank you for your reply.

We remove this row.

Affiliate-B commented on ads/_config.js:
```diff
@@ -20,8 +20,88 @@
+ * URLs to connect to for a given ad type.
+ * No need to add URLs that have same hosts as in the adPrefetch.
+ * This MUST be kept in sync with actual implementation.
+=======
```
We resolve conflict.

Affiliate-B commented:
Thank you for your reply.
We resolve conflict.

lannka commented on ads/_config.js:
```diff
@@ -20,8 +20,83 @@
- *   prefetch: string|array
+ * @const {!Object<string, (string|!Array<string>)>}
+ */
+export const adPrefetch = {
```
can you revert this? probably from a bad merge

zhouyx commented on ads/_config.js:
```diff
@@ -98,6 +173,11 @@ export const adConfig = {
     prefetch: 'https://s.d.adup-tech.com/jsapi',
   },
 
+  affiliateb: {
```
agree with @lannka. only this piece of code should be added to this file

zhouyx commented:
Also could you reply to googlebot. 
```
Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.
```

zhouyx commented on ads/_config.js:
```diff
@@ -98,6 +98,11 @@ export const adConfig = {
+    prefetch: 'https://track.affiliate-b.com/amp/a.js',
+    renderStartImplemented: true,
+  },
+  
```
to pass linter check, please remove trailing space.
you can run `gulp lint` locally to test

zhouyx commented:
@Affiliate-B could you reply "I signed it!" to googlebot here.

Affiliate-B commented:
I signed it!

2016-10-06 9:49 GMT+09:00 Yuxuan Zhou <notifications@github.com>:

> @Affiliate-B <https://github.com/Affiliate-B> could you reply "I signed
> it!" to googlebot here.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ampproject/amphtml/pull/5223#issuecomment-251839922>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AU-6PTv_YfcPFDGt4e0UgcRW3Y7jcemIks5qxEWcgaJpZM4KGgFo>
> .
>



-- 
◆業界最速！報酬30日サイクル振込＆外税8％上乗支払のアフィリエイトB◆
[image: 株式会社フォーイット] <http://www.for-it.co.jp/> *渡部 陽介*

YOUSUKE WATANABE
プロダクト・ディベロップメント部
部長

*株式会社フォーイット* <http://www.for-it.co.jp/>
〒150-0044
東京都渋谷区円山町3-6
E・スペースタワー7F <http://goo.gl/maps/Q3Wsg>

TEL:03-5728-5865
FAX:03-5728-4480
watanabe@for-it.co.jp
[image: AffiliateB] <https://www.affiliate-b.com/> [image: スマイルモール]
<http://sumamo.com/> [image: アフィリエイトB facebookページ]
<https://www.facebook.com/forit.affiliateb>

Affiliate-B commented:
Thank you for your help!
We think that the conflict has been resolved.

lannka commented:
To get the CLA robot happy, @Affiliate-B you might try to follow its instruction: https://github.com/ampproject/amphtml/pull/5223#issuecomment-249564276

Affiliate-B commented:
We changed email address on github.

zhouyx commented:
looks great!

zhouyx commented:
@Affiliate-B 
Thanks for the PR. merged.

Affiliate-B commented:
Thank you for giving me good advice.
Thank you so much!

Affiliate-B commented:
We tested the tag of affiliate B on my web page, but we get errors.

errors :
Type "affiliateb" is not supported in amp-ad    error.js:68
Uncaught Error: failed to build: amp-ad#1: Type "affiliateb" is not supported in amp-ad​​​ _reported_    log.js:243

What should I do?

zhouyx commented:
@Affiliate-B Don't worry. The PR has not made to prod yet. It should be alive in prod by the end of next week. 
If you need to test now, The code is in canary right, you can go to [experiment page](https://cdn.ampproject.org/experiments.html) and opt in our `dev-channel`.  Once opted in you should be able to see the code.

Affiliate-B commented:
OK. I got it.
Thanks a lot！
